butane_cli/src/lib.rs

```diff
@@ -92,6 +93,125 @@ pub fn make_migration(base_dir: &Path, name: Option<&String>) -> Result<()> {
+    }
+    if old.is_pk() != new.is_pk() {
+        println!("  pk: {} -> {}", old.is_pk(), new.is_pk());
+    }
+    if old.is_auto() != new.is_auto() {
+        println!("  auto: {} -> {}", old.is_auto(), new.is_auto());
+    }
+    if old.nullable() != new.nullable() {
+        println!("  nullable: {} -> {}", old.nullable(), new.nullable());
+    }
+    if old.unique() != new.unique() {
+        println!("  unique: {} -> {}", old.unique(), new.unique());
+    }
+    if old.default() != new.default() {
+        println!("  default: {:?} -> {:?}", old.default(), new.default());
+    }
+    if old.reference() != new.reference() {
+        let old = match old.reference() {
+            Some(ARef::Literal(reference)) => {
+                format!("{}.{}", reference.table_name(), reference.column_name())
+            }
+            None => "None".to_string(),
+            Some(ARef::Deferred(_)) => panic!(),
+        };
+        let new = match new.reference() {
+            Some(ARef::Literal(reference)) => {
+                format!("{}.{}", reference.table_name(), reference.column_name())
+            }
+            None => "None".to_string(),
+            Some(ARef::Deferred(_)) => panic!(),
+        };
+        println!("  references: {} -> {}", old, new);
+    }
+    Ok(())
+}
+
+/// Print description of a list of [`Operation`].
+pub fn print_ops(ops: Vec<Operation>) -> Result<()> {
+    if ops.is_empty() {
+        println!("No changes");
+        return Ok(());
+    }
+    for op in &ops {
+        use Operation::*;
+        match op {
+            AddTable(table) | AddTableIfNotExists(table) => {
+                println!("New table {}", table.name);
+                for column in &table.columns {
+                    println!("  {} -> {:?}", column.name(), column.typeid()?);
+                }
+            }
+            AddTableConstraints(_) => {}
+            RemoveTable(name) => {
+                println!("Remove table {}", name);
+            }
+            AddColumn(table_name, column) => {
+                println!("New column {table_name}.{}", column.name());
+            }
+            RemoveColumn(table_name, column_name) => {
+                println!("Remove column {table_name}.{column_name}");
+            }
+            ChangeColumn(table_name, old, new) => {
+                let column_name = old.name();
+                // Rename currently isn't supported.
+                // https://github.com/Electron100/butane/issues/89
+                assert_eq!(column_name, new.name());
+                println!("Change column {}.{column_name}", table_name);
+                print_column_diff(old, new)?;
+            }
+        }
+    }
+    Ok(())
+}
+
+/// Print a description of the current migration.
+pub fn describe_current_migration(base_dir: &Path) -> Result<()> {
+    let mut ms = get_migrations(base_dir)?;
+    let to_db = ms.current().db()?;
+    let mut from_db = ADB::new();
```

Comment:
[nit] mutable is unnecessary here, the `ADB::new()` can be in an else clause (at line 181)